boot/bootchain.go

```diff
@@ -255,3 +256,35 @@ func bootAssetsToLoadChains(assets []bootAsset, kernelBootFile bootloader.BootFi
 	}
 	return chains, nil
 }
+
+func bootChainsFromFile(path string) (pbc predictableBootChains, err error) {
+	inf, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("cannot open existing boot chains state file: %v", err)
+	}
+	defer inf.Close()
+	if err := json.NewDecoder(inf).Decode(&pbc); err != nil {
+		return nil, fmt.Errorf("cannot read boot chains data: %v", err)
+	}
+	return pbc, nil
+}
+
+func bootChainsToFile(pbc predictableBootChains, path string) error {
+	if err := os.MkdirAll(filepath.Dir(path), 0755); err != nil {
+		return fmt.Errorf("cannot create device fde state directory: %v", err)
+	}
+	outf, err := osutil.NewAtomicFile(path, 0600, 0, osutil.NoChown, osutil.NoChown)
+	if err != nil {
+		return fmt.Errorf("cannot create a temporary boot chains file: %v", err)
+	}
+	// becomes noop when the file is committed
+	defer outf.Cancel()
+
+	if err := json.NewEncoder(outf).Encode(pbc); err != nil {
```

Comment:
also I forgot, it's never a good forward compatible idea to encode an array directly, we probably want to encode as:

{"boot-chains": ..}